### ISSUE TITLE
The option for `]test` could run `rr`

### ISSUE THREAD

**Issue by vtjnash** [JuliaLang/Pkg.jl#2591]
Being able to specify `--bug-report=rr-local` or `--bug-report=rr` would be very useful. Then reproducing a test failure is as simple as `Base.require(Main, :BugReporting).replay("/file/path")` without needing to mess with temporary environments and such

**Comment by DilumAluthge** [comment]
I think that you can provide command-line args in the `julia_args` keyword argument to the `Pkg.test` function.

**Comment by DilumAluthge** [comment]
Try this and let me know if it works?

```julia
Pkg.test(; julia_args = ["--bug-report=rr-local"])
```

**Comment by vtjnash** [comment]
That appears to run the REPL instead of the tests

**Comment by vtjnash** [comment]
It seems to work, but is pretty long:

`
while true; do ./julia -e 'using Pkg; Pkg.test("SPICE", julia_args=["--bug-report=rr-local", "/home/vtjnash/.julia/packages/SPICE/CjGzi/test/runtests.jl"])' || exit ; done
`

**Comment by vtjnash** [comment]
Oh, though perhaps that is also stripping off `--project`?

**Comment by DilumAluthge** [comment]
Hmmm. In theory, you shouldn't need to worry about passing the `runtests` file or anything like that.

If you do this, what happens? You just get the REPL?

```julia
julia> Pkg.test(; julia_args = ["--bug-report=rr-local"])
```

**Comment by KristofferC** [comment]
Yes you get a REPL but the reason for that is that you also get a REPL when doing `julia --bug-report=rr-local --eval 'print("hello")'` as well (which is similar to how Pkg invokes julia to run the tests).

**Comment by DilumAluthge** [comment]
Oh, interesting. Is there any way to do `julia --bug-report=rr-local my_script.jl` or `julia --bug-report=rr-local --eval 'print("hello")'` and not get a REPL (i.e. have Julia exit once it is done).

If that feature were added, then it would be easy enough to do e.g.:
```julia
Pkg.test("SPICE"; julia_args = ["--bug-report=rr-local-no-repl"])
```

**Comment by vtjnash** [comment]
I have solved this with a small helper script:

```
$ cat bin-rr/julia 
#!/bin/sh
rr record --nested=detach $(dirname $(dirname $0))/bin/julia "$@"

$ JULIA_BINDIR=~+/bin-rr ./bin/julia
```
